#T_base_sensor:
# - [1, 0, 0, 0.3]
# - [ 0, 1, 0, 0]
# - [ 0,  0, 1, 0]
# - [ 0.0, 0.0, 0.0, 1.0]

 
# General Params
verbosity: 3
global_frame_name: world
label_path: /home/lukas/Documents/PanopticMapping/Data/3RScan/labels.csv
change_detection_interval: -1.0
visualization_interval: -1.0

tsdf_integrator:  
  type: projective # naive, projective, projective_multi_tsdf
  verbosity: 1
  
  # projective args
  # rio rendered
#  width: 960
#  height: 540
#  vx: 488.945
#  vy: 269.186
#  fx: 756.525
#  fy: 755.581
  
  # rio original
  width: 224
  height: 172
  vx: 113.696
  vy: 85.4004
  fx: 176.523
  fy: 240.667
    
  max_range: 5  # m
  min_range: 0.1
  use_weight_dropoff: true
  sparsity_compensation_factor: 1.0
  max_weight: 10000
  interpolation_method: bilinear  # nearest, bilinear, adaptive, semantic
  foreign_rays_clear: true

id_tracker:
  type: ground_truth
  verbosity: 2
  input_is_mesh_id: false
  instance_voxel_size: 0.03
  background_voxel_size: 0.08
  unknown_voxel_size: 0.08
  freespace_voxel_size: 0.3
  voxels_per_side: 16
  
tsdf_registrator:
  verbosity: 1
  min_voxel_weight: 0.000001
  error_threshold: -2 # m, negative = #voxel_sizes
  match_rejection_points: 50
  match_rejection_percentage: 0.1
  match_acceptance_points: 50
  match_acceptance_percentage: 0.1
  normalize_by_voxel_weight: true
  
visualization:
  verbosity: 3
  
  # Submaps
  color_mode: color # submaps, instances, color, normals, classes, change
  visualization_mode: active  # all, active 
  submap_color_discretization: 20
  visualize_mesh: true
  visualize_tsdf_blocks: true
  visualize_free_space: true
  visualize_bounding_volumes: true
  include_free_space: true
  mesh_min_weight: 0.000001
  
  # Planning
  visualize_slice: true
  planning_slice_resolution: 0.08
  planning_slice_height: -1.0


  

